src/Contract/QueryAuctions.purs

```diff
@@ -10,47 +10,61 @@ import Contract.Monad (Contract)
 import Contract.Scripts (validatorHash)
 import Contract.Time (POSIXTime)
 import Contract.Transaction (TransactionOutput)
+import Contract.Utxos (utxosAt)
 import Contract.Value (CurrencySymbol)
 import Contract.Value (valueOf) as Value
 import Contract.Wallet (ownPaymentPubKeyHash)
-import Control.Error.Util (bool)
 import Control.Monad.Maybe.Trans (MaybeT(MaybeT), runMaybeT)
 import Control.Monad.Trans.Class (lift)
+import Ctl.Internal.Plutus.Types.Transaction (_output)
 import Data.Array (mapMaybe) as Array
+import Data.Lens ((^.))
+import Data.Map (toUnfoldable) as Map
+import Data.Newtype (modify)
 import Data.Validation.Semigroup (isValid) as V
 import HydraAuctionOffchain.Contract.MintingPolicies (auctionMetadataTokenName)
 import HydraAuctionOffchain.Contract.PersonalOracle (mkPersonalOracle)
 import HydraAuctionOffchain.Contract.Types
-  ( ActorRole
+  ( ActorRole(ActorRoleBidder, ActorRoleSeller)
   , AssetClass(AssetClass)
   , AuctionActor(AuctionActor)
   , AuctionFilters
   , AuctionInfo(AuctionInfo)
+  , AuctionInfoExtended
   , AuctionTerms
+  , Utxo
+  , mkAuctionInfoExtended
   , validateAuctionTerms
   )
 import HydraAuctionOffchain.Contract.Validators (mkAuctionMetadataValidator)
-import HydraAuctionOffchain.Helpers (getInlineDatum, getTxOutsAt)
+import HydraAuctionOffchain.Helpers (getInlineDatum)
 
--- | Queries all currently existing auctions at the auction metadata validator address,
--- | filtering out invalid entries.
-queryAuctions :: AuctionFilters -> Contract (Array AuctionInfo)
+-- | Queries all currently existing auctions at the auction metadata
+-- | validator address, filtering out invalid entries.
+queryAuctions :: AuctionFilters -> Contract (Array AuctionInfoExtended)
 queryAuctions filters =
   case (unwrap filters).myRole of
     Just actorRole ->
       fromMaybe mempty <$> queryOwnAuctions actorRole
     Nothing -> do
       auctionMetadataAddr <- flip scriptHashAddress Nothing <<< validatorHash <$>
         mkAuctionMetadataValidator
-      getTxOutsAt auctionMetadataAddr
+      utxosAt auctionMetadataAddr
         <#> Array.mapMaybe getValidAuctionInfo
+        <<< Map.toUnfoldable
   where
-  getValidAuctionInfo :: TransactionOutput -> Maybe AuctionInfo
-  getValidAuctionInfo txOut = do
+  getValidAuctionInfo :: Utxo -> Maybe AuctionInfoExtended
+  getValidAuctionInfo (oref /\ txOutWithRefScript) = do
     auctionInfo@(AuctionInfo { auctionId, auctionTerms }) <- getInlineDatum txOut
-    bool Nothing (Just auctionInfo)
-      (validAuctionTerms auctionTerms && validAuctionId auctionId)
+    case validAuctionTerms auctionTerms && validAuctionId auctionId of
+      true ->
+        Just $ mkAuctionInfoExtended auctionInfo $ Just oref
+      false ->
+        Nothing
     where
+    txOut :: TransactionOutput
+    txOut = txOutWithRefScript ^. _output
```

Comment:
Just curious: can we use dot-syntax here or not?

Reply:
We should, fixed it in https://github.com/mlabs-haskell/hydra-auction-offchain/pull/30/commits/e65e9af08c1e93f7d2282d19ed0176c4cd1e50a0